lightning/src/routing/router.rs

```diff
@@ -1520,14 +1520,13 @@ where L::Target: Logger {
 	for path in selected_paths.iter_mut() {
 		let mut shadow_ctlv_expiry_delta_offset: u32 = 0;
 
-		// Choose starting point for the random walk
-		let mut cur_node_id = if network_nodes.contains_key(&payee_node_id) {
-			payee_node_id 
-		} else {
-			assert_eq!(payee_node_id, NodeId::from_pubkey(&path.last().unwrap().as_ref().unwrap().pubkey));
-			assert!(path.len() > 1);
-			NodeId::from_pubkey(&path.iter().rev().next().unwrap().as_ref().unwrap().pubkey)
-		};
+		// Choose the last publicly known node as the starting point for the random walk
```

Comment:
Using the last publicly known node as the starting point for the random walk is better than the solution before, but still not sufficient: how should we deal with entirely private paths? Better not add any random offset at all in this case?

Reply:
Hmm, good question, I guess we can always add some fixed value? Like rand(1-3)*144 or something?